addon/index.js

```diff
@@ -494,7 +494,7 @@ export function changeset(obj, validateFn = defaultValidatorFn, validationMap =
         this._deleteKey(ERRORS, key);
 
         if (!isEqual(oldValue, value)) {
-          set(changes, key, value);
+          changes[key] = value;
```

Comment:
I'm not quite sure why `Ember.set` fails to update value. I think changing property directly might be okay since we call `this.notifyPropertyChange(CHANGES);` later.